src/Http/Requests/StoreUserDisk.php

```diff
@@ -53,4 +53,14 @@ public function getDiskOptions()
 
         return $options;
     }
+
+    public function withValidator($validator)
+    {
+        $validator->after(function ($validator) {
+            // Different user may have equal disk names, but disk names of one user must be unqiue
+            if (UserDisk::where(['user_id' => $this->user()->id, 'name' => $this->input('name')])->get()->isNotEmpty()) {
```

Comment:
You can also use the slightly more efficient query:

```suggestion
            if (UserDisk::where(['user_id' => $this->user()->id, 'name' => $this->input('name')])->exists()) {
```